src/common/hooks/useStateSafely.ts

```diff
@@ -0,0 +1,21 @@
+import type { Dispatch, SetStateAction } from 'react';
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+export default function useStateSafely<S>(initialState: S): [S, Dispatch<SetStateAction<S>>] {
```

Comment:
We usually use arrow functions, why mixing the style with `function` keyword ?

Reply:
A: This module exports a single default function. Using a lambda here (as in many other places in our codebase) would demand declaring it in one statement and then adding an additional statement for the exported member.

Reply:
Ok, Im fine with it